### ISSUE TITLE
Secondary mouse button up events lost during mouse capture

### ISSUE THREAD

**Issue by NkD** [glfw/glfw#954]
GLFW version 3.3.0
LWJGL version 3.1.1
OS: Windows 10 (edited)

Replication:
Window size 1400*800 (decorated, windowed, not fullscreen)
```
GLFWCursorPosCallback - xpos = 1395.0, ypos = 437.0
GLFWMouseButtonCallback - action = GLFW_PRESS, button = GLFW_MOUSE_BUTTON_LEFT
GLFWMouseButtonCallback - action = GLFW_PRESS, button = GLFW_MOUSE_BUTTON_RIGHT
GLFWCursorPosCallback - xpos = 1396.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1397.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1398.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1399.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1400.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1401.0, ypos = 437.0 <-  outside window, but still recieving event because mouse down (it's ok)
GLFWCursorPosCallback - xpos = 1402.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1403.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1404.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1405.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1406.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1407.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1408.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1409.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1410.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1411.0, ypos = 437.0
GLFWCursorPosCallback - xpos = 1412.0, ypos = 437.0
GLFWMouseButtonCallback - action = GLFW_RELEASE, button = GLFW_MOUSE_BUTTON_RIGHT <- release one button outside window
```
Right now I still have GLFW_MOUSE_BUTTON_LEFT in state GLFW_PRESS, but GLFWCursorPosCallback stop recieve event. And if I release left mouse button right now a don't get event : GLFWMouseButtonCallback : GLFW_RELEASE -> GLFW_MOUSE_BUTTON_LEFT)

**Comment by elmindreda** [comment]
You are only guaranteed to receive cursor position events while the cursor is over the window client area. Only on macOS are cursor position events generated when the cursor is outside the window and you are not dragging.

**Comment by elmindreda** [comment]
Which OS are you on?

**Comment by NkD** [comment]
Windows 10

**Comment by NkD** [comment]
Imagine some logic in my App listener which prepare some logic on mouse press and finish it on release. But I can not ensure that it always arrives opposite event.
With this GFLW behavior I have event orphans.. press event sometime without release event.

**Comment by NkD** [comment]
Main problem in GLFW is, that any GLFW_RELEASE event stop dragging process, even when release button is not the one that started the dragging. This behavior broke oppositeness press and release events.

**Comment by elmindreda** [comment]
Ah, the button up messages are being eaten by the end of cursor capture. I'll look into it.

**Comment by elmindreda** [comment]
This should be fixed now with dd96d0ac93dcdf77557a24535cd24afba31f6a71. Thank you for reporting this!
